volatility3/framework/layers/intel.py

```diff
@@ -501,3 +507,85 @@ class WindowsIntel32e(WindowsMixin, Intel32e):
 
     def _translate(self, offset: int) -> Tuple[int, int, str]:
         return self._translate_swap(self, offset, self._bits_per_register // 2)
+
+
+class LinuxMixin(Intel):
+    @functools.cached_property
+    def register_mask(self) -> int:
```

Comment:
This is exposing a lot of additional fields and I just want to make sure the balance is right between things that the OS might actually need to know about, and things that really should only be used internally by the intel layer.  At the moment the guts feel like they're on the outside and I want to be sure every single method we expose is worthwhile.

I'd also change TODOs that we know aren't accurate to FIXMEs please, otherwise there'll be no urgency of any kind of improve them and this will all just be adding extra lines for no reason.

Reply:
I made all the methods introduced by the LinuxMixin internal.

> I'd also change TODOs that we know aren't accurate to FIXMEs please, otherwise there'll be no urgency of any kind of improve them and this will all just be adding extra lines for no reason.

Sorry, I'm not sure which TODOs you're referring to here. Could you please point them out to me?

Reply:
I thought there was one that said "we know this isn't right", but I can only find the revisit once SME is supported, which is fine as a TODO.